### DISCUSSION TITLE
why is `afterNavigate` called on mount?

### DISCUSSION THREAD

**Discussion by mayank99** [sveltejs/kit#10719] [Q&A] [answered]
hi! i'm trying to perform an action whenever a navigation happens.

`afterNavigate` seems like the perfect api for this, except it also invokes the callback when the page first loads, which is not at all what i want.

from the docs of `afterNavigate`:
> A lifecycle function that runs the supplied callback _when the current component mounts_, and also whenever we navigate to a new URL.

this feels strange, as a navigation is when you go from one page to another, rather than when the page is mounted (that's what `onMount` is for).

if this behavior is intentional, then is there a workaround to skip the callback on first load?

**Comment by aloker** [comment, marked as answer]
You can use the`type` property of the object passed to the callback to determine why the callback was invoked. When the value is "enter" you know it's the initial page hydration and you can skip your custom post-navigation logic.

See https://kit.svelte.dev/docs/types#public-types-afternavigate

**Reply by mayank99** [reply]
oh wow, i would have never thought to look in the types reference.

this is exactly what i needed, thanks!
